test/mainline_test.go

```diff
@@ -339,6 +349,24 @@ func removeInstallResourceCR(c client.Client, name string, ctx context.Context)
 	Expect(err).NotTo(HaveOccurred())
 	err = c.Delete(ctx, instance)
 	Expect(err).NotTo(HaveOccurred())
+	// Need to wait here for Installation resource to be fully deleted prior to cancelling the context
+	// which will in turn terminate the operator. Race conditions can occur otherwise that will leave the
+	// Installation resource intact while the operator is no longer running which will result in test failures
+	// that try to create an Installation resource of their own
+	Eventually(func() error {
+		installation := &operator.Installation{
```

Comment:
would it make sense to encap this into a helper function rather than duplicate it here and also above?

Reply:
Yeah good call. There's actually a helper function that already exists so I've repurposed it a bit and added a timeout parameter, then replaced the relevant blocks in my diff to use it.